packages/hasura-auth-js/src/hasura-auth-client.ts

```diff
@@ -132,6 +133,45 @@ export class HasuraAuthClient {
     )
   }
 
+  /**
+   * Use `nhost.auth.connectProvider` to connect a social authentication provider to an existing user account
+   *
+   * @example
+   * ### Connect an authentication provider to an existing user account
+   * ```ts
+   * nhost.auth.connectProvider({
+   *   provider: 'github
+   *   options: {
+   *    redirectTo: window.location.href
+   *   }
+   * })
+   * ```
+   *
+   * @docs https://docs.nhost.io/reference/javascript/auth/connect-provider
+   */
+  async connectProvider(
+    params: SignInWithProviderParams
+  ): Promise<SignInResponse & { providerUrl?: string; provider?: string }> {
```

Comment:
should we change the response type? I am going to change the backend to just respond with "OK" and given that you are hardcoding the response `session: null, mfa: null` it feels like we need a different type